infra/modules/azure_api_management/locals.tf

```
@@ -1,18 +1,7 @@
 locals {
-  # General
-  location_map = {
-    "italynorth"         = "itn",
-    "westeurope"         = "weu",
-    "germanywestcentral" = "gwc"
-  }
-  location_short   = lookup(local.location_map, var.environment.location, "neu")
-  project          = "${var.environment.prefix}-${var.environment.env_short}-${local.location_short}"
-  domain           = var.environment.domain == null ? "-" : "-${var.environment.domain}-"
-  apim_name_prefix = "${local.project}${local.domain}${var.environment.app_name}"
-
   apim = {
-    name           = "${local.apim_name_prefix}-apim-${var.environment.instance_number}"
-    autoscale_name = var.tier == "l" ? "${local.apim_name_prefix}-apim-autoscale-${var.environment.instance_number}" : null
+    name           = "${module.naming_convention.prefix}-apim-${module.naming_convention.suffix}"
+    autoscale_name = var.tier == "l" ? "${module.naming_convention.prefix}-apim-autoscale-${module.naming_convention.suffix}" : null
```

`autoscale` should be abbreviated `as` (not part of this PR)
```suggestion
    autoscale_name = var.tier == "l" ? "${module.naming_convention.prefix}-apim-as-${module.naming_convention.suffix}" : null
```